pallets/loans/src/migrations.rs

```diff
@@ -0,0 +1,89 @@
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::*;
+
+/// This migration nukes all storages from the pallet individually.
+pub struct Migration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for Migration<T> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		// The current state from altair when this migration should be done has only one element
+		// in each store.
+		ensure!(
+			PoolToLoanNftClass::<T>::iter_values().count() == 1,
+			"Err PoolToLoanNftClass"
+		);
+		ensure!(
+			LoanNftClassToPool::<T>::iter_values().count() == 1,
+			"Err LoanNftClassToPool"
+		);
+		ensure!(
+			NextLoanId::<T>::iter_values().count() == 1,
+			"Err NextLoanId"
+		);
+		ensure!(Loan::<T>::iter_values().count() == 1, "Err Loan");
+		ensure!(
+			ActiveLoans::<T>::iter_values().count() == 1,
+			"Err ActiveLoans"
+		);
+		ensure!(
+			ClosedLoans::<T>::iter_values().count() == 1,
+			"Err ClosedLoans"
+		);
+		ensure!(PoolNAV::<T>::iter_values().count() == 1, "Err PoolNAV");
+		ensure!(
+			PoolWriteOffGroups::<T>::iter_values().count() == 1,
+			"Err PoolWriteOffGroups "
+		);
+
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		let _ = PoolToLoanNftClass::<T>::clear(1, None);
```

Comment:
Should: Execute only for specific storage version, see below:

```rust
if Pallet::<T>::on_chain_storage_version() == StorageVersion::new(0) {
       // snip
}
```

Reply:
I understand this check, but not sure if it works for this "migration" that is not a migration 😆. (See comment above)

Reply:
You are right. I have to admit I was blanking out on the fact that `pallet-loans` co-exists with `pallet-loans-ref` and will be deprecated. So no need to add versioning!

Reply:
Don't worry! Sometimes I'm also confused with this mess and I'm the person in change of this 🤣